### PR TITLE
test: Removes IDs From Datasource Directory Tests

### PR DESCRIPTION
**Pull request by andre-dossinger**
## Purpose
* Removes fixed IDs from tests to enable landscape independent fixture recording. 

## Does this introduce a breaking change?
```
[ ] Yes
[ x ] No
```

## Pull Request Type

What kind of change does this Pull Request introduce?
```
[ ] Bugfix
[ ] Feature
[ x ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test

* Test the code via automated test

```bash
go test ./...
```

## What to Check

Verify that the following are valid:

* Automated tests are executed successfully

## Checklist for reviewer
The following organizational tasks must be completed before merging this PR:

* [x] The PR is assigned to the Terraform project and a status is set (typically "in review").
* [x] The PR has the matching labels assigned to it.
* [x] The PR has a milestone assigned to it.
* [x] If the PR closes an issue, the issue is referenced.
* [x] Possible follow-up items are created and linked.

### CODE REVIEW

**Comment by andre-dossinger** [comment]
Prerequisite for #59

File: internal/provider/datasource_directory_test.go [hunk 1]
```diff
@@ -22,33 +22,33 @@ func TestDataSourceDirectory(t *testing.T) {
...
+						resource.TestMatchResourceAttr("data.btp_directory.uut", "id", regexpValidUUID),
 						resource.TestCheckResourceAttrSet("data.btp_directory.uut", "created_by"),
-						resource.TestCheckResourceAttr("data.btp_directory.uut", "created_date", "2023-05-16T08:39:33Z"),
+						//resource.TestCheckResourceAttr("data.btp_directory.uut", "created_date", "2023-05-16T08:39:33Z"),
 						resource.TestCheckResourceAttr("data.btp_directory.uut", "description", "Please don't modify. This is used for integration tests."),
 						resource.TestCheckResourceAttr("data.btp_directory.uut", "labels.#", "0"),
 						resource.TestMatchResourceAttr("data.btp_directory.uut", "last_modified", regexpValidRFC3999Format),
 						resource.TestCheckResourceAttr("data.btp_directory.uut", "name", "integration-test-dir-static"),
-						resource.TestCheckResourceAttr("data.btp_directory.uut", "parent_id", "03760ecf-9d89-4189-a92a-1c7efed09298"),
+						//resource.TestCheckResourceAttr("data.btp_directory.uut", "parent_id", "03760ecf-9d89-4189-a92a-1c7efed09298"),
```
Comment (lechnerc77) [inline internal/provider/datasource_directory_test.go]: Does it make sense to make a check against UUID => TestMatchResourceAttr like for ID per se?

File: internal/provider/datasource_directory_test.go [hunk 2]
```diff
@@ -22,33 +22,33 @@ func TestDataSourceDirectory(t *testing.T) {
...
 				},
 				{ // security enabled directory
-					Config: hclProviderFor(user) + hclDatasourceDirectory("uut_security_enabled", "05368777-4934-41e8-9f3c-6ec5f4d564b9"),
+					Config: hclProviderFor(user) + hclDatasourceDirectory("uut_security_enabled", "integration-test-dir-se-static"),
 					Check: resource.ComposeAggregateTestCheckFunc(
-						resource.TestCheckResourceAttr("data.btp_directory.uut_security_enabled", "id", "05368777-4934-41e8-9f3c-6ec5f4d564b9"),
+						resource.TestMatchResourceAttr("data.btp_directory.uut_security_enabled", "id", regexpValidUUID),
 						resource.TestCheckResourceAttrSet("data.btp_directory.uut_security_enabled", "created_by"),
-						resource.TestCheckResourceAttr("data.btp_directory.uut_security_enabled", "created_date", "2023-05-16T08:46:24Z"),
+						//resource.TestCheckResourceAttr("data.btp_directory.uut_security_enabled", "created_date", "2023-05-16T08:46:24Z"),
```
Comment (lechnerc77) [inline internal/provider/datasource_directory_test.go]: Can be completely removed

File: internal/provider/datasource_directory_test.go [hunk 3]
```diff
@@ -22,33 +22,33 @@ func TestDataSourceDirectory(t *testing.T) {
...
 						resource.TestCheckResourceAttr("data.btp_directory.uut_security_enabled", "description", "Please don't modify. This is used for integration tests."),
 						resource.TestCheckResourceAttr("data.btp_directory.uut_security_enabled", "labels.#", "0"),
 						resource.TestMatchResourceAttr("data.btp_directory.uut_security_enabled", "last_modified", regexpValidRFC3999Format),
 						resource.TestCheckResourceAttr("data.btp_directory.uut_security_enabled", "name", "integration-test-dir-se-static"),
-						resource.TestCheckResourceAttr("data.btp_directory.uut_security_enabled", "parent_id", "03760ecf-9d89-4189-a92a-1c7efed09298"),
+						//resource.TestCheckResourceAttr("data.btp_directory.uut_security_enabled", "parent_id", "03760ecf-9d89-4189-a92a-1c7efed09298"),
```
Comment (lechnerc77) [inline internal/provider/datasource_directory_test.go]: See comment above concerning pattern check

File: internal/provider/datasource_directory_test.go [hunk 4]
```diff
@@ -22,33 +22,33 @@ func TestDataSourceDirectory(t *testing.T) {
...
 						resource.TestCheckResourceAttr("data.btp_directory.uut_security_enabled", "state", "OK"),
-						resource.TestCheckResourceAttr("data.btp_directory.uut_security_enabled", "subdomain", "05368777-4934-41e8-9f3c-6ec5f4d564b9"),
+						//resource.TestCheckResourceAttr("data.btp_directory.uut_security_enabled", "subdomain", "05368777-4934-41e8-9f3c-6ec5f4d564b9"),
```
Comment (lechnerc77) [inline internal/provider/datasource_directory_test.go]: Check can be removed

**Review by lechnerc77 — CHANGES_REQUESTED** [review]
Minor adjustments needed - see comments

File: internal/provider/datasource_directory_test.go (hunk 1 above)
Comment (andre-dossinger) [reply internal/provider/datasource_directory_test.go]: I agree. Actually I forgot to adjust the commented out sections. hanks for the hint.

**Review by lechnerc77 — APPROVED** [review]
lgtm

**Pull request merged**